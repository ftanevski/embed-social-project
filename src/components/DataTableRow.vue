<template>
    <div class="row">
        <div class="colContainer">
            <data-table-row-columns
                v-for="(key, value) in row"
                :title="value"
                :text="key"
                :key="key"
            >
            </data-table-row-columns>
            <div class="col-md-2">
                <p class="text-muted">Action</p>
                <button class="btn btn-default btn-sm push-5-r">
                    <i class="fa fa-edit"></i>
                </button>
                <button class="btn btn-default btn-sm push-5-r">
                    <i class="fa fa-eye"></i>
                </button>
                <button class="btn btn-default btn-sm push-5-r">
                    <i class="fa fa-trash"></i>
                </button>
            </div>
        </div>
    </div>
</template>

<script>
import DataTableRowColumns from './DataTableRowColumns';

export default {
    name: 'DataTableRow',
    components: {
        DataTableRowColumns
    },
    props: {
		row: {
			type: Object,
			required: true,
            description: 'Objects containing form response information'
		}
	},
};
</script>

<style>
.colContainer {
    align-items: center;
    display: flex;
    padding: 20px;
}
</style>