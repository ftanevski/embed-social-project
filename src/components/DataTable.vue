<template>
    <modal 
        v-if="responseModalActive"
        @close-modal="responseModalStatus"
    >
        <template #header>
            <h2 class="font-w600 push-20-l">{{ modalResponse.name }}'s Response</h2>
        </template>
        <template #content>
            <data-table-row
                :row="modalResponse"
                :vertical="true"
                @open-modal="responseModalStatus"
            >
            </data-table-row>
        </template>
    </modal>
    <data-filter
        v-if="filterModalActive"
        @change-filter-modal-status="changeFilterModalStatus">
    </data-filter>
    <data-table-header
        @update-posts="updateNumOfPosts"
        @filter-modal="changeFilterModalStatus"
    >
    </data-table-header>
    <div class="col border">
        <data-table-row
            v-for="(response, index) in numberOfPosts"
            :row="response"
            :key="index"
            @open-modal="responseModalStatus"
            :vertical="false"
		>
        </data-table-row>
    </div>
    <div class="button-container justify-content-center mt-5">
        <button
            class="btn btn-primary"
            v-if="showButton"
            @click="showMorePosts"
            data-testid="load-more-test"
        >
        Load More
        </button>
    </div>
</template>
 
<script>
import DataTableRow from './DataTableRow';
import DataTableHeader from './DataTableHeader';
import DataFilter from './DataFilter';
import Modal from './Modal';
 
export default {
    name: 'DataTable',
    components: {
        DataTableRow,
        DataTableHeader,
        DataFilter,
        Modal
    },
    data() {
        return {
            displayedPosts: 5,
            postsToLoad: 5,
            responseModalActive: false,
            filterModalActive: false,
            modalResponse: false
        }
    },
    props: {
        responses: {
            type: Array,
            default: () => [],
            description: 'Array of objects that contain data for each row in the table.'
        }
    },
    computed: {
        numberOfPosts() {
            return this.responses.slice(0, this.displayedPosts);
        },
        showButton() {
            return this.displayedPosts <= this.responses.length;
        }
    },
    methods: {
        showMorePosts() {
            this.displayedPosts += this.postsToLoad;
        },
        updateNumOfPosts(option) {
            this.displayedPosts = option;
            this.postsToLoad = option;
        },
        responseModalStatus(row) {
            this.responseModalActive = !this.responseModalActive;
            this.modalResponse = row;
        },
        changeFilterModalStatus() {
            this.filterModalActive = !this.filterModalActive;
        }
    }
};
</script>
 
<style>
.button-container {
    margin: 30px;
}
</style>