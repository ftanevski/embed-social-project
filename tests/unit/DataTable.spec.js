import { mount } from '@vue/test-utils';
import DataTable from '../../src/components/DataTable';
import DataTableRow from '../../src/components/DataTableRow';
import formattedData from '../../src/services/formatSubmissions';

describe('DataTable', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(DataTable);
	});

	test('Snapshot matches', () => {
		expect(wrapper.element).toMatchSnapshot();
	});

	test('DataTable renders all rows from component', () => {
		const dataTableRow = wrapper.findAllComponents(DataTableRow);
		expect(dataTableRow).toHaveLength(formattedData.length);
	});
});