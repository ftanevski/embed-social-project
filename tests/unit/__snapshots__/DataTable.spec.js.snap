// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable Matches the snapshot 1`] = `
<div
  data-v-app=""
>
  
  <!--v-if-->
  <!--v-if-->
  <div
    class="row header-row"
  >
    <div
      class="dropdown"
    >
      <select
        class="form-control"
        data-testid="dropdown-test"
      >
        
        <option
          class="select-options"
          data-testid="dropdown-options-test"
          value="5"
        >
          5
        </option>
        <option
          class="select-options"
          data-testid="dropdown-options-test"
          value="10"
        >
          10
        </option>
        <option
          class="select-options"
          data-testid="dropdown-options-test"
          value="15"
        >
          15
        </option>
        <option
          class="select-options"
          data-testid="dropdown-options-test"
          value="20"
        >
          20
        </option>
        <option
          class="select-options"
          data-testid="dropdown-options-test"
          value="25"
        >
          25
        </option>
        
      </select>
    </div>
    <div
      class="left-button"
    >
      <button
        class="btn btn-secondary"
      >
        Filter Submissions
      </button>
    </div>
    <div
      class="right-button"
    >
      <button
        class="btn btn-primary"
      >
        Create Submission
      </button>
    </div>
  </div>
  <div
    class="col border"
  >
    
    <div
      class="table-row"
    >
      
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Name
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Ivana Trajkova
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Email
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          ivana_tr@example.com
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Recommendation
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          yes
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Feedback
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas non enim gravida ante feugiat fermentum in vitae sem. Mauris ligula ante, finibus quis convallis eget, auctor vehicula massa. Nunc arcu lectus, viverra ac enim eu, suscipit fermentum sapien. Nam maximus facilisis enim sit amet tempor. Curabitur at euismod augue, et aliquet elit. Mauris dui libero, consequat sit amet lacus sit amet, vulputate interdum tellus. Morbi condimentum porttitor ornare.
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Date
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          13/04/2021
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Form
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Form 1
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Form type
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          popup
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Status
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          pending
        </div>
      </div>
      
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
        >
          Action
        </p>
        <button
          class="btn btn-default btn-sm push-5-r"
        >
          <i
            class="fa fa-edit"
          />
        </button>
        <button
          class="btn btn-default btn-sm push-5-r"
          data-testid="preview-button-test"
        >
          <i
            class="fa fa-eye"
          />
        </button>
        <button
          class="btn btn-default btn-sm push-5-r"
        >
          <i
            class="fa fa-trash"
          />
        </button>
      </div>
    </div>
    <div
      class="table-row"
    >
      
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Name
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Ivona Atanasova
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Email
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          ivana_ar@example.com
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Recommendation
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          maybe
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Feedback
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas non enim gravida ante feugiat fermentum in vitae sem. Mauris ligula ante, finibus quis conva
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Date
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          10/11/2020
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Form
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Form 2
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Form type
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          page
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Status
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          approved
        </div>
      </div>
      
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
        >
          Action
        </p>
        <button
          class="btn btn-default btn-sm push-5-r"
        >
          <i
            class="fa fa-edit"
          />
        </button>
        <button
          class="btn btn-default btn-sm push-5-r"
          data-testid="preview-button-test"
        >
          <i
            class="fa fa-eye"
          />
        </button>
        <button
          class="btn btn-default btn-sm push-5-r"
        >
          <i
            class="fa fa-trash"
          />
        </button>
      </div>
    </div>
    <div
      class="table-row"
    >
      
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Name
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Jonathan
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Email
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          jonathan@gmail.com
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Recommendation
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          maybe
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Feedback
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          My neighbor Karly has one of these. She works as a gambler and she says it looks tall. I saw one of these in Spratly Islands and I bought one. works okay.
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Date
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          16/03/2021
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Form
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Form 3
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Form type
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          page
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Status
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          approved
        </div>
      </div>
      
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
        >
          Action
        </p>
        <button
          class="btn btn-default btn-sm push-5-r"
        >
          <i
            class="fa fa-edit"
          />
        </button>
        <button
          class="btn btn-default btn-sm push-5-r"
          data-testid="preview-button-test"
        >
          <i
            class="fa fa-eye"
          />
        </button>
        <button
          class="btn btn-default btn-sm push-5-r"
        >
          <i
            class="fa fa-trash"
          />
        </button>
      </div>
    </div>
    <div
      class="table-row"
    >
      
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Name
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Nicole
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Email
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          nicole@yahoo.com
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Recommendation
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          no
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Feedback
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          The box this comes in is 4 yard by 5 inch and weights 12 pound! i use it profusely when i'm in my garage.
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Date
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          02/04/2021
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Form
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Form 4
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Form type
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          popup
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Status
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          pending
        </div>
      </div>
      
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
        >
          Action
        </p>
        <button
          class="btn btn-default btn-sm push-5-r"
        >
          <i
            class="fa fa-edit"
          />
        </button>
        <button
          class="btn btn-default btn-sm push-5-r"
          data-testid="preview-button-test"
        >
          <i
            class="fa fa-eye"
          />
        </button>
        <button
          class="btn btn-default btn-sm push-5-r"
        >
          <i
            class="fa fa-trash"
          />
        </button>
      </div>
    </div>
    <div
      class="table-row"
    >
      
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Name
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Mike
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Email
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          mike@hotmail.com
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Recommendation
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          yes
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Feedback
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          This Coca-Cola works really well. It wildly improves my baseball by a lot.
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Date
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          13/04/2021
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Form
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          Form 5
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Form type
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          email
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
          data-testid="title-test"
        >
          Status
        </p>
        <div
          class="font-w600"
          data-testid="text-test"
        >
          approved
        </div>
      </div>
      
      <div
        class="col-md-2"
      >
        <p
          class="text-muted"
        >
          Action
        </p>
        <button
          class="btn btn-default btn-sm push-5-r"
        >
          <i
            class="fa fa-edit"
          />
        </button>
        <button
          class="btn btn-default btn-sm push-5-r"
          data-testid="preview-button-test"
        >
          <i
            class="fa fa-eye"
          />
        </button>
        <button
          class="btn btn-default btn-sm push-5-r"
        >
          <i
            class="fa fa-trash"
          />
        </button>
      </div>
    </div>
    
  </div>
  <div
    class="button-container justify-content-center mt-5"
  >
    <button
      class="btn btn-primary"
      data-testid="load-more-test"
    >
       Load More 
    </button>
  </div>
  
</div>
`;
